openedx/core/djangoapps/content/block_structure/tasks.py

```diff
@@ -24,26 +24,28 @@
 @task(
     default_retry_delay=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_DEFAULT_RETRY_DELAY'],
     max_retries=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_MAX_RETRIES'],
+    bind=True,
 )
-def update_course_in_cache(course_id):
+def update_course_in_cache(self, course_id):
     """
     Updates the course blocks (in the database) for the specified course.
     """
-    _call_and_retry_if_needed(course_id, api.update_course_in_cache, update_course_in_cache)
+    _call_and_retry_if_needed(course_id, api.update_course_in_cache, update_course_in_cache, self.request.id)
 
 
 @task(
     default_retry_delay=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_DEFAULT_RETRY_DELAY'],
     max_retries=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_MAX_RETRIES'],
+    bind=True,
 )
-def get_course_in_cache(course_id):
+def get_course_in_cache(self, course_id):
     """
     Gets the course blocks for the specified course, updating the cache if needed.
     """
-    _call_and_retry_if_needed(course_id, api.get_course_in_cache, get_course_in_cache)
+    _call_and_retry_if_needed(course_id, api.get_course_in_cache, get_course_in_cache, self.request.id)
```

Comment:
For what it's worth, when I was doing work similar to this and needed the task id, `self.task_id` didn't exist. I inspected `self` and there was simply no attribute with that name.